src/Core/Content/Category/Service/CategoryBreadcrumbBuilder.php

```diff
@@ -109,6 +123,76 @@ public function getProductSeoCategory(ProductEntity $product, SalesChannelContex
         return null;
     }
 
+    private function getCategoryByReferer(?Request $request, ProductEntity $product, SalesChannelContext $context): ?CategoryEntity
+    {
+        if ($request === null || $request->headers->has('referer') === false) {
```

Comment:
i mean this will be an issue with the http cache. The header is not considered in the cache key. Which results that the side will be cached with the first "referer|null" and will never be updated